var/spack/repos/builtin/packages/neurodamus-core/package.py

```diff
@@ -1,103 +1,112 @@
     """Library of channels developed by Blue Brain Project, EPFL"""
 
     homepage = "ssh://bbpcode.epfl.ch/sim/neurodamus-core"
     git      = "ssh://bbpcode.epfl.ch/sim/neurodamus-core"
 
-    version('develop', git=git, branch='master', clean=False)
-    version('2.8.0', git=git, tag='2.8.0', clean=False)
-    version('2.7.3', git=git, tag='2.7.3', clean=False)
-    version('2.7.2', git=git, tag='2.7.2', clean=False)
-    version('2.7.0', git=git, tag='2.7.0', clean=False)
-    version('2.6.0', git=git, tag='2.6.0', clean=False)
-    version('2.5.0', git=git, tag='2.5.0', clean=False)
-    version('2.4.3', git=git, tag='2.4.3', clean=False)
-    version('2.4.1', git=git, tag='2.4.1', clean=False)
-    version('2.3.4', git=git, tag='2.3.4', clean=False)
-    version('2.3.3', git=git, tag='2.3.3', clean=False)
-    version('2.2.1', git=git, tag='2.2.1', clean=False)
-
-    variant('python', default=False, description="Enable Python Neurodamus")
-    variant('common', default=False, description="Merge in synapse mechanisms hoc & mods")
-
-    # Attempt to support building
-    depends_on('neuron~binary+python', when='+common')
-
-    # Neurodamus py is currently an extension to core
-    resource(name='pydamus',
-             git='ssh://bbpcode.epfl.ch/sim/neurodamus-py',
-             when='+python',
-             destination='resources')
+    version('develop', branch='master', clean=False)
+    version('2.9.1', tag='2.9.1', clean=False)
+    version('2.9.0', tag='2.9.0', clean=False)
+    version('2.8.0', tag='2.8.0', clean=False)
+    version('2.7.3', tag='2.7.3', clean=False)
+    version('2.7.2', tag='2.7.2', clean=False)
+    version('2.7.0', tag='2.7.0', clean=False)
+    version('2.6.0', tag='2.6.0', clean=False)
+    version('2.5.0', tag='2.5.0', clean=False)
+    version('2.4.3', tag='2.4.3', clean=False)
+    version('2.4.1', tag='2.4.1', clean=False)
+    version('2.3.4', tag='2.3.4', clean=False)
+    version('2.3.3', tag='2.3.3', clean=False)
+    version('2.2.1', tag='2.2.1', clean=False)
+
+    variant('mpi',    default=True,  description="Enable MPI support")
+    variant('common', default=False, description="Bring in common synapse mechanisms")
+    variant('hdf5',   default=True,  description="Enable old Hdf5 reader")
+    variant('reportinglib', default=True, description="Enable ReportingLib")
+    variant('synapsetool',  default=True, description="Enable SynapseTool reader (for edges)")
+    variant('mvdtool',      default=True, description="Enable MVDTool reader (for nodes)")
+
+    # NOTE: Several variants / dependencies come from SimModel
+    depends_on("mpi",  when='+mpi', type=('build', 'run'))  # dont link
+    depends_on("hdf5+mpi", when='+hdf5+mpi')
+    depends_on("hdf5~mpi", when='+hdf5~mpi')
+    depends_on('reportinglib',         when='+reportinglib')
+    depends_on('reportinglib+profile', when='+reportinglib+profile')
+    depends_on('synapsetool',          when='+synapsetool')
```

Comment:
why reportinglib is dependency with neurodamus-core? IIRC, neurodamus-core was not building auxiliary mod files (report, hdf5 etc related).

similar question for synapsetool, mvdtool etc.

Reply:
humm... I guess I failed to explain the idea. neurodamus-core will have the mods compiled, since we can load two libraries and have the core independent from the model. We hopefully in the future can drop the mixed-together `neurodamus-model`